### PR TITLE
ci: rewrite java ci

### LINKED ISSUE #25: Autogenerated Gradle module is incorrect
Currently, only the first Gradle module (a Gradle alternative to POMs) that is completed by the CI will be published on Maven. Typically, that's OSX. As a result, the Gradle module will show OSX as the only artifact.

This is not the end of the world - [codemp-intellij](https://github.com/hexedtech/codemp-intellij) is unaffected by this because it specifies the classifier it's interested in anyway. However, this may potentially compromise usability of the library by third parties since, because of this, the default artifact you receive if you don't specify a classifier is the OSX one.

Fixing this may require a considerable rework of how our Maven CI works, but I'm not sure of how to approach the problem. I don't especially want a GitHub actions box spinning 3 docker instances.

### LINKED ISSUE #40: Auto-release on Maven Central will fail if the grace period is exhausted
Typically Mac is the fastest CI to compile, with Linux as a close second. Windows lags behind significantly, sometimes taking more than 5 minutes, which is the "grace period" Maven Central gives us to upload remaining stuff of a release after the first batch.

This will result in the Windows version sometimes not being uploaded at all, as it has unfortunately happened with version 0.7.3.

Combined with #25, it makes a good argument for rewriting the Java CI altogether.

### PR DESCRIPTION
**Pull request by zaaarf**
this way no matter how long windows takes, all will be correctly deployed.
should fix #25 and fix #40

### CODE REVIEW

**Comment by zaaarf** [comment]
The fact that this works upsets me more than if it didn't.

File: .github/workflows/java.yml [hunk 1]
```diff
@@ -23,30 +23,49 @@ jobs:
...
       - uses: gradle/actions/setup-gradle@v4
         with:
-          gradle-version: "8.10" # Quotes required to prevent YAML converting to number
-      - run: gradle build
+          gradle-version: "8.10"
+      - run: gradle windowsJar
+        working-directory: dist/java
+      - run: gradle macosJar
+        working-directory: dist/java
+      - run: gradle linuxJar
```
Comment (alemidev) [inline .github/workflows/java.yml]:
```suggestion
```
i dont think these are needed, can we remove them to make the publishing process faster?

**Review by alemidev — CHANGES_REQUESTED** [review]
we should probably remove unneeded build steps

**Review by alemidev — APPROVED** [review]

**Comment by zaaarf** [comment]
In the future I'd like to make the `build.gradle` less messy, but this is good enough for now.

**Pull request merged**